### PR TITLE
cli command to set scheduled MRENCLAVE

### LINKED ISSUE #1626: Add command-line to set the scheduled MRENCLAVE
### Context

More of a ulitity tool

---
:heavy_check_mark: Please set appropriate **labels** and **assignees** if applicable.

### PR DESCRIPTION
**Pull request by kziemianek**
closes: #1626 

Adds cli command to set scheduled MRENCLAVE.
Only for dev/testing purposes.

### CODE REVIEW

File: tee-worker/app-libs/stf/src/trusted_call.rs [hunk 1]
```diff
@@ -718,6 +722,12 @@ where
 				)));
 				Ok(())
 			},
+			TrustedCall::set_scheduled_mrenclave(_, bn, mrenclave) => {
+				GLOBAL_SCHEDULED_ENCLAVE
+					.update(bn, mrenclave)
+					.map_err(|_| StfError::SetScheduledMrEnclaveFailed)?;
```
Comment (Kailai-Wang) [inline tee-worker/app-libs/stf/src/trusted_call.rs]: Here we lose the original root cause - maybe simply log it?

File: tee-worker/app-libs/stf/src/trusted_call.rs [hunk 2]
```diff
@@ -718,6 +722,12 @@ where
 				)));
 				Ok(())
 			},
+			TrustedCall::set_scheduled_mrenclave(_, bn, mrenclave) => {
```
Comment (Kailai-Wang) [inline tee-worker/app-libs/stf/src/trusted_call.rs]: It's slightly risky to leave out the first account checker, although we know (internally) it's used for dev purposes only. Imagine the code is accidentally published to a production enclave.

I suggest adding at least `is_root` checker, but this is a minor problem.

File: tee-worker/cli/src/trusted_base_cli/commands/litentry/set_scheduled_mrenclave.rs [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+	pub(crate) fn run(&self, cli: &Cli, trusted_cli: &TrustedCli) {
+		let root = get_pair_from_str(trusted_cli, "//Alice");
+
+		let (mrenclave, shard) = get_identifiers(trusted_cli);
+		let nonce = get_layer_two_nonce!(root, cli, trusted_cli);
+
+		let top: TrustedOperation = TrustedCall::set_scheduled_mrenclave(
+			root.public().into(),
+			self.bn,
+			mrenclave_from_base58(&self.mrenclave),
```
Comment (Kailai-Wang) [inline tee-worker/cli/src/trusted_base_cli/commands/litentry/set_scheduled_mrenclave.rs]: Maybe we should expect a hex string here - just to be consistent with the extrinsic in teerex

**Review by Kailai-Wang — APPROVED** [review]
Looks good - I approved it.

There're a few small suggestions, feel free to adjust accordingly (or not) and merge it.

**Review by Kailai-Wang — APPROVED** [review]
Looks good - I approved it.

There're a few small suggestions, feel free to adjust accordingly (or not) and merge it.

File: tee-worker/cli/src/trusted_base_cli/commands/litentry/set_scheduled_mrenclave.rs (hunk 3 above)
Comment (kziemianek) [reply tee-worker/cli/src/trusted_base_cli/commands/litentry/set_scheduled_mrenclave.rs]: done

File: tee-worker/app-libs/stf/src/trusted_call.rs (hunk 1 above)
Comment (kziemianek) [reply tee-worker/app-libs/stf/src/trusted_call.rs]: done

File: tee-worker/app-libs/stf/src/trusted_call.rs (hunk 2 above)
Comment (kziemianek) [reply tee-worker/app-libs/stf/src/trusted_call.rs]: done

**Pull request merged**